mkt/site/utils.py

```diff
@@ -677,9 +678,16 @@ def get_icon_url(base_url_format, obj, size,
         return '{path}/{name}'.format(path=static_url('ICONS_DEFAULT_URL'),
                                       name=default_format.format(size=size))
     else:
-        # [1] is the whole ID, [2] is the directory.
-        split_id = re.match(r'((\d*?)\d{1,3})$', str(obj.pk))
         # If we don't have the icon_hash set to a dummy string ("never"),
         # when the icon is eventually changed, icon_hash will be updated.
         suffix = obj.icon_hash or 'never'
+
+        if storage_is_remote():
+            # We don't care about base_url_format, the storage provides the url
+            # for a given path. We assume AWS_QUERYSTRING_AUTH is False atm.
```

Comment:
let's add this to the config with a comment that turning it on breaks assumptions